WordPress/gutenberg
# Template Parts: Add rename template part command

Related: 
- https://github.com/WordPress/gutenberg/issues/52651

## What?

Adds a rename template part command.

_Note: Whether or not the template part commands and modals should be managed on their own or alongside patterns is still a bit unclear at the moment and likely to change based on feedback._

_Refactoring or consolidating the modal components will be looked into via a follow-up._

## Why?

This is a step toward parity between the template part and pattern functionality in the site editor. It also lays the ground work for allowing quick duplication of a template part via a duplicate command.

## How?

- Adds a new site editor command for renaming template parts
- Uses the original rename modal from before the patterns page was switched to dataviews (simpler rebase. consolidating components will be a follow-up).

## Testing Instructions

1. Navigate to Appearance > Editor > Patterns
2. Navigate to a template part category
3. Select a non-customized theme provided template part
4. Confirm that you do not get the rename template part _command_ 
5. Customize the template part
6. Confirm that you now get the rename _command_ and it works as expected
7. Duplicate the template part
8. Confirm that the rename _command_ is present and works

## Screenshots or screencast 


https://github.com/WordPress/gutenberg/assets/60436221/76848f1f-5c8f-46b2-8287-1b6c3cab200d



<details>
<summary>Original incarnation of this pre dataviews for patterns</summary>

## What?

Adds a rename template part command and simplifies the pattern page's `RenameMenuItem` component.

**Note: Whether or not the template part commands and modals should be managed on their own or alongside patterns is still a bit unclear at the moment and likely to change based on feedback.**

## Why?

This is a step toward parity between the template part and pattern functionality in the site editor. It also lays the ground work for allowing quick duplication of a template part via a duplicate command.

## How?

- Extracts a `RenameTemplatePartModal` from the pattern page's `RenameMenuItem`
- Updates `RenameMenuItem` to use the new rename modals for patterns and template parts
- Adds a new command for template parts

## Testing Instructions

1. Navigate to Appearance > Editor > Patterns
2. Confirm renaming a user-created pattern still works as expected
3. Navigate to a template part category
4. Select a non-customized theme provided template part
5. Confirm that you do not get the rename template part _command_ 
6. Customize the template part
7. Confirm that you now get the rename _command_ and it works as expected
8. Duplicate the template part
9. Confirm that the rename _command_ is present and works

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/60436221/820d36e3-fe7d-4ceb-9dca-54a5ad8017d3

</details>
